ee/server/services/ddp-streamer/Server.ts

```diff
@@ -67,18 +85,34 @@ export class Server extends EventEmitter {
 	async subscribe(client: Client, packet: IPacket): Promise<void> {
 		try {
 			if (!this._subscriptions.has(packet.name)) {
-				throw new Error(`Subscription '${packet.name}' doesn't exist`);
+				// TODO should not be MethorError
+				throw new MethodError(404, `Subscription '${packet.name}' not found`);
 			}
 			const fn = this._subscriptions.get(packet.name);
 			if (!fn) {
-				throw new Error('subscription not found');
+				throw new MethodError(404, `Subscription '${packet.name}' not found`);
 			}
 
 			const publication = new Publication(client, packet, this);
 			const [eventName, options] = packet.params;
 			await fn.call(publication, eventName, options);
-		} catch (error) {
+		} catch (error: any) {
 			this.nosub(client, packet, error.toString());
+
+			if (error instanceof ClientSafeError) {
+				return this.nosub(client, packet, error.toJSON());
+			}
+
+			// default errors are logged to the console and redacted from the client
+			logger.error({ msg: 'Subscribe error', error });
+
+			return this.nosub(client, packet, {
```

Comment:
Why not pass the error instance and instead of accepting any record to return as error?

Reply:
good idea!